snap-envi-reader/src/main/java/org/esa/snap/dataio/envi/EnviProductWriter.java

```diff
@@ -26,25 +28,74 @@
 import java.io.RandomAccessFile;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.prefs.Preferences;
 
 /**
  * The product writer for ENVI products.
- *
  */
 public class EnviProductWriter extends AbstractProductWriter {
 
+    private final static String SYSPROP_USE_CACHE = "snap.dataio.writer.envi.useCache";
+
     protected File _outputDir;
     protected File _outputFile;
     private Map _bandOutputStreams;
     private boolean _incremental = true;
 
+    private boolean useCache;
+    private WriteCache writeCache;
+
     /**
      * Construct a new instance of a product writer for the given ENVI product writer plug-in.
      *
      * @param writerPlugIn the given ENVI product writer plug-in, must not be <code>null</code>
      */
     public EnviProductWriter(ProductWriterPlugIn writerPlugIn) {
         super(writerPlugIn);
+
+        final Preferences preferences = Config.instance().preferences();
+        useCache = preferences.getBoolean(SYSPROP_USE_CACHE, true);
+        writeCache = new WriteCache();
```

Comment:
Why is a WriteCache instantiated even if useCache is false?

Reply:
resolved - create conditional